core/src/main/java/org/elasticsearch/index/mapper/core/KeywordFieldMapper.java

```diff
@@ -0,0 +1,308 @@
+                if (!omitNormsSet && fieldType.boost() == 1.0f) {
+                    fieldType.setOmitNorms(true);
+                }
+                if (!indexOptionsSet) {
+                    fieldType.setIndexOptions(IndexOptions.DOCS);
+                }
+            }
+            setupFieldType(context);
+            if (context.indexCreatedVersion().onOrAfter(Version.V_3_0_0)
+                    && fieldType.indexOptions().compareTo(IndexOptions.DOCS_AND_FREQS_AND_POSITIONS) >= 0) {
+                throw new MapperParsingException("Field [" + name
+                        + "] of type [keyword] only supports indexing docs and freqs, but got [index_options="
+                        + indexOptionToString(fieldType.indexOptions()) + "]");
+            }
+            KeywordFieldMapper fieldMapper = new KeywordFieldMapper(
+                    name, fieldType, defaultFieldType, ignoreAbove,
+                    context.indexSettings(), multiFieldsBuilder.build(this, context), copyTo);
+            fieldMapper.includeInAll(includeInAll);
+            return fieldMapper;
+        }
+    }
+
+    public static class TypeParser implements Mapper.TypeParser {
+        @Override
+        public Mapper.Builder parse(String name, Map<String, Object> node, ParserContext parserContext) throws MapperParsingException {
+            KeywordFieldMapper.Builder builder = keywordField(name);
+            parseField(builder, name, node, parserContext);
+            if (parserContext.indexVersionCreated().before(Version.V_3_0_0)) {
+                parseTermVectorSettings(builder, name, node, parserContext);
+            }
+
+            /* TODO: Add support for analyzed keyword fields with analyzers that generate a single token
+            if (node.containsKey("analyzer")) {
+                final String analyzerName = node.remove("analyzer").toString();
+                final NamedAnalyzer analyzer = parserContext.analysisService().analyzer(analyzerName);
+                if (analyzer == null) {
+                    throw new MapperParsingException("analyzer [" + analyzerName + "] not found for field [" + name + "]");
+                }
+                builder.indexAnalyzer(analyzer);
+                builder.searchAnalyzer(analyzer);
+            }*/
+
+            for (Iterator<Map.Entry<String, Object>> iterator = node.entrySet().iterator(); iterator.hasNext();) {
+                Map.Entry<String, Object> entry = iterator.next();
+                String propName = Strings.toUnderscoreCase(entry.getKey());
+                Object propNode = entry.getValue();
+                if (propName.equals("null_value")) {
+                    if (propNode == null) {
+                        throw new MapperParsingException("Property [null_value] cannot be null.");
+                    }
+                    builder.nullValue(propNode.toString());
+                    iterator.remove();
+                } else if (propName.equals("search_quote_analyzer")
```

Comment:
Does that mean if you create a `keyword` analyzer on an old index, it will allow all these settings inherited from StringFieldMapper? Or am I missing a check for this somewhere?